src/Illuminate/Contracts/Filesystem/Filesystem.php

```diff
@@ -63,9 +59,6 @@ public function put($path, $contents, $options = []);
      * @param  resource  $resource
      * @param  array  $options
      * @return bool
-     *
-     * @throws \InvalidArgumentException If $resource is not a file handle.
-     * @throws \Illuminate\Contracts\Filesystem\FileExistsException
```

Comment:
What happens on failure now? Just returns false? What about the invalid arg exception from before? Seems odd to no longer be able to tell the difference between a failed write and the code being invoked wrong by a dev?

Reply:
I've explained this in the main comment of the PR.